pkg/controller/machine/controller.go

```diff
@@ -154,6 +154,17 @@ func (r *ReconcileMachine) Reconcile(request reconcile.Request) (reconcile.Resul
 			return reconcile.Result{}, err
 		}
 
+		if m.Status.NodeRef != nil {
+			klog.Infof("Deleting node %q", m.Status.NodeRef.Name)
+			var node corev1.Node
+			key := client.ObjectKey{Namespace: "", Name: m.Status.NodeRef.Name}
+			if err := r.Client.Get(ctx, key, &node); err != nil {
```

Comment:
You might even go with `r.Client.Get(ctx, client.ObjectKey{Name: m.Status.NodeRef.Name}, &node)`